leetcodePython/BinarySearch/bs_1508.py

```diff
@@ -0,0 +1,58 @@
+import heapq
+from typing import List
+
+
+class SolutionBruteForce:
+    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
+        sums = []
+        for start in range(len(nums)):
+            curr_sum = 0
+            for end in range(start, len(nums)):
+                curr_sum += nums[end]
+                sums.append(curr_sum)
+        sums.sort()
+        return sum(sums[left - 1:right]) % (10 ** 9 + 7)
+
+
+class SolutionSlidingWindowBS:
+    @staticmethod
+    def _count_and_sum_target(nums: List[int], n: int, target: int) -> tuple[int, int]:
+        curr_sum = 0
+        left = 0
+        window_sum = 0
+        count = 0
+        total_sum = 0
+        for right in range(n):
+            curr_sum += nums[right]
+            #
+            window_sum += nums[right] * (right - left + 1)
+            while curr_sum > target:
+                window_sum -= curr_sum
```

Comment:
Sum for this specific window is bigger than target, we have to exclude this window